scylla-cql/src/frame/response/result.rs

```diff
@@ -1090,6 +1090,71 @@ mod tests {
         }
     }
 
+    #[cfg(feature = "num-bigint-04")]
+    #[test]
+    fn test_varint04() {
+        use num_bigint_04::ToBigInt;
+
+        struct Test<'a> {
+            value: num_bigint_04::BigInt,
+            encoding: &'a [u8],
+        }
+
+        /*
+            Table taken from CQL Binary Protocol v4 spec
+
+            Value | Encoding
+            ------|---------
+                0 |     0x00
+                1 |     0x01
+              127 |     0x7F
+              128 |   0x0080
+              129 |   0x0081
+               -1 |     0xFF
+             -128 |     0x80
+             -129 |   0xFF7F
+        */
+        let tests = [
+            Test {
+                value: 0.to_bigint().unwrap(),
+                encoding: &[0x00],
+            },
+            Test {
+                value: 1.to_bigint().unwrap(),
+                encoding: &[0x01],
+            },
+            Test {
+                value: 127.to_bigint().unwrap(),
+                encoding: &[0x7F],
+            },
+            Test {
+                value: 128.to_bigint().unwrap(),
+                encoding: &[0x00, 0x80],
+            },
+            Test {
+                value: 129.to_bigint().unwrap(),
+                encoding: &[0x00, 0x81],
+            },
+            Test {
+                value: (-1).to_bigint().unwrap(),
+                encoding: &[0xFF],
+            },
+            Test {
+                value: (-128).to_bigint().unwrap(),
+                encoding: &[0x80],
+            },
+            Test {
+                value: (-129).to_bigint().unwrap(),
+                encoding: &[0xFF, 0x7F],
+            },
+        ];
```

Comment:
I don't like the amount of copy paste here... Can we extract these test cases from `test_varint03` and `test_varint04` somewhere else, e.g. create a function that returns the test cases? The type of `value` could be changed to `Vec<u8>` probably.